dds/DCPS/security/CommonUtilities.cpp

```diff
@@ -79,6 +79,78 @@ void set_security_error(DDS::Security::SecurityException& ex,
   set_security_error(ex, code, minor_code, full.c_str());
 }
 
+OPENDDS_STRING ctk_to_dds_string(const CryptoTransformKind& keyKind)
+{
+  if (!keyKind[0] && !keyKind[1] && !keyKind[2]) {
+    switch (keyKind[3]) {
+    case CRYPTO_TRANSFORMATION_KIND_NONE:
+      return "CRYPTO_TRANSFORMATION_KIND_NONE";
+    case CRYPTO_TRANSFORMATION_KIND_AES128_GMAC:
+      return "CRYPTO_TRANSFORMATION_KIND_AES128_GMAC";
+    case CRYPTO_TRANSFORMATION_KIND_AES128_GCM:
+      return "CRYPTO_TRANSFORMATION_KIND_AES128_GCM";
+    case CRYPTO_TRANSFORMATION_KIND_AES256_GMAC:
+      return "CRYPTO_TRANSFORMATION_KIND_AES256_GMAC";
+    case CRYPTO_TRANSFORMATION_KIND_AES256_GCM:
+      return "CRYPTO_TRANSFORMATION_KIND_AES256_GCM";
+    }
+  }
+  OPENDDS_STRING rv("Unknown CryptoTransformKind ");
+  const size_t i = rv.size();
+  rv.resize(i + 11);
+  std::sprintf(&rv[i], "%.2x %.2x %.2x %.2x",
+    keyKind[0], keyKind[1], keyKind[2], keyKind[3]);
+  return rv;
+}
+
+OPENDDS_STRING ctki_to_dds_string(const CryptoTransformKeyId& keyId)
+{
+  OPENDDS_STRING rv('X', 12);
+  std::sprintf(&rv[0], "%.2x %.2x %.2x %.2x",
+    keyId[0], keyId[1], keyId[2], keyId[3]);
+  rv.resize(11);
+  return rv;
+}
+
+OPENDDS_STRING to_dds_string(const KeyOctetSeq& keyData)
+{
+  size_t l = keyData.length() * 2;
+  // For Spaces
+  if (l > 2) {
+    l += (l / 2) - 1;
+  }
+  OPENDDS_STRING rv;
+  rv.reserve(l);
+  for (size_t i = 0; i < keyData.length(); i++) {
```

Comment:
Several places in here where a single good generic "bytes_to_hex_string" function would serve us well (see comment above).